values/resolve.go

```diff
@@ -0,0 +1,329 @@
+package values
+
+import (
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/semantic"
+)
+
+// Resolver represents a value that can resolve itself.
+// Resolving is the action of capturing the scope at function declaration and
+// replacing any identifiers with static values from the scope where possible.
+// TODO(nathanielc): Improve implementations of scope to only preserve values
+// in the scope that are referrenced.
+type Resolver interface {
```

Comment:
All of the code in this file was moved from `interpreter/interpreter.go` with only minimal changes so that it can compile.